phalcon/cache/backend/file.zep

```diff
@@ -121,32 +121,45 @@ class File extends Backend implements BackendInterface
 
 			let frontend = this->_frontend;
 
-			/**
-			 * Take the lifetime from the frontend or read it from the set in start()
-			 */
-			if !lifetime {
-				let lastLifetime = this->_lastLifetime;
-				if !lastLifetime {
-					let ttl = (int) frontend->getLifeTime();
-				} else {
-					let ttl = (int) lastLifetime;
-				}
-			} else {
-				let ttl = (int) lifetime;
-			}
-
-			let modifiedTime = (int) filemtime(cacheFile);
+            let cachedContent = json_decode(file_get_contents(cacheFile), true);
+            /**
+             * Take the lifetime from the frontend or read it from the set in start()
+             * if the cachedContent is not a valid array
+             */
+            if !lifetime {
+                if this->isValidArray(cachedContent, "lifetime") {
+            		let ttl = (int) cachedContent["lifetime"];
+            	} else {
+            	    let lastLifetime = this->_lastLifetime;
+            	    if !lastLifetime {
+            	        let ttl = (int) frontend->getLifeTime();
+            	    } else {
+            		    let ttl = (int) lastLifetime;
+            		}
+            	}
+            } else {
+            	let ttl = (int) lifetime;
+            }
+            if !this->isValidArray(cachedContent, "created") {
+            	let createdTime = (int) filemtime(cacheFile);
+            } else {
+            	let createdTime = (int) cachedContent["created"];
+            }
```

Comment:
Usually my IDE is using tabs as well, should be the .zep extension. Will add another commit fixing the indents.